docs/rows.md

```diff
@@ -0,0 +1,152 @@
+client.rows.query(plan=plan)
+```
+
+Optic supports many different types of queries and operations; please
+[see the documentation]((https://docs.marklogic.com/guide/app-dev/OpticAPI#id_35559)) for further information on 
+much more powerful and flexible queries than shown in these examples, which are intended solely for demonstration of 
+how to submit an Optic query.
+
+
+## SQL queries
+
+MarkLogic supports [SQL queries against views](https://docs.marklogic.com/guide/sql). SQL queries can be submitted 
+via the `sql` argument:
+
+```
+client.rows.query(sql="select * from example.musician order by lastName")
+```
+
+This will return a JSON object that captures each of the matching rows along with definitions 
+for each column. See the section below on choosing an output format for controlling how data is returned.
+
+## SPARQL queries
+
+MarkLogic supports the [SPARQL query language](https://www.w3.org/TR/sparql11-query/), allowing for 
+[SPARQL queries to be run against views](https://docs.marklogic.com/guide/semantics/semantic-searches#id_94155), 
+similar to Optic and SQL. SPARQL queries can be submitted via the `sparql` argument:
+
+```
+sparql = "PREFIX musician: <http://marklogic.com/column/example/musician/> SELECT * WHERE {?s musician:lastName ?lastName} ORDER BY ?lastName"
+client.rows.query(sparql=sparql)
+```
+
+This will return a JSON object that captures each of the matching rows along with definitions 
+for each column. See the section below on choosing an output format for controlling how data is returned. 
+
+## GraphQL queries
+
+MarkLogic supports [GraphQL queries](https://docs.marklogic.com/REST/POST/v1/rows/graphql) to be run against views. 
+A GraphQL query can be submitted via the `graphql` argument:
+
+```
+client.rows.query(graphql="query myQuery { example_musician { lastName firstName dob } }")
+```
+
+This will return a JSON object containing the matching rows. MarkLogic will only return a JSON object for GraphQL
+queries; the `format` argument described below will not have any impact when submitting a GraphQL query.
+
+## Choosing an output format
+
+The [MarkLogic REST endpoint for querying rows](https://docs.marklogic.com/REST/POST/v1/rows) supports several options
+for how data is returned via the `format` parameter. The `client.rows.query` function allows for an output format to be 
+selected via a `format` argument. The following table defined the possible values:
+
+| `format` value | Output format | 
+| --- | --- |
+| `json` | JSON object with keys of 'columns' and 'rows'. |
+| `xml` | XML document defining the columns and rows. |
+| `csv` | CSV text with the first row defining the columns. |
+| `json-seq` | A [line-delimited JSON sequence](https://datatracker.ietf.org/doc/html/rfc7464) with the first row defining the columns. |
+| `mixed` | TODO Seems like we should remove this as it does the same thing as "return_response". |
```

Comment:
@BillFarber  See my note here - thoughts on removing this? I doubt it would ever be used, and `return_response=True` would be the more consistent way to just get the original multipart/mixed response back.

Reply:
If 'mixed' just has the same affect as "return_response=true", then I would remove it.
However, would it work and be useful to use Client.process_multipart_mixed_response(response) to pre-process requests with the 'mixed' format?

Reply:
Not that I know of yet, and in fact, one reason I'd like to move the eval stuff out of client.py is so that the `process_multipart...` method isn't yet public. While it may be useful in the future as a public method, we don't know that yet, and I lean towards making things non-public when in doubt (as opposed to what I did with e.g. ml-app-deployer).